app/components/forms/events/view/order-form.js

```diff
@@ -20,11 +23,47 @@ export default Component.extend(FormMixin, {
       }
     };
   },
+  typeList: computed(function() {
+    return orderBy(fieldTypes, 'type');
+  }),
+  normalFields: computed('data.customForms', function() {
+    return this.data.customForms.filter(field => {
+      return !field.isCustomQuestion;
+    });
+  }),
   actions: {
     submit(data) {
-      this.onValid(() => {
-        this.save(data);
+      let fields = data.customForms;
+      let incompleteFields = fields.filter(function(field) {
+        return ((!field.fieldIdentifier || !field.type || field.fieldIdentifier === '') && field.isCustomQuestion);
       });
+      if (incompleteFields.length > 0) {
+        this.notify.error(this.l10n.t('Existing fields need to be completed before new items can be added.'));
+      } else {
+        this.onValid(() => {
+          this.save(data);
+        });
+      }
+    },
+    deleteQuestion(field) {
+      field.destroyRecord();
```

Comment:
destroyRecord returns a promise. Why are you ignoring it ?

Reply:
@niranjan94 I am now using a simple `await` for now, should I encapsulate it in a `try`-`catch`?